File: csci308/lab/CppB/Ref.java
public class Ref{

    public boolean mybool = true;
    private int myint = 8; 

    public void print() {
        System.out.printf("%b %d\n", this.mybool, this.myint);
    }

    public void change(int i) {
        this.myint = i;
    } 

    public static void main(String[] args) {
      Ref obj1 = new Ref(); 
      Ref obj2 = obj1;

      obj1.print();
      obj2.print();
      obj1.change(42); // change a member data to 42
      obj1.print();
      obj2.print();
    }
}

File: csci308/lab/CppB/Ref.cpp
#include<iostream>

// Ref class
// public bool mybool
// private in myint
// public print() and change()

using namespace std;
class Ref {
public:
  bool mybool;
  Ref();
  void print();
  void change(int i);

private:
  int myint;
};

// mybool = 1
// myint = 8


Ref::Ref() {
  this->mybool = true;
  this->myint = 8;
}

void Ref::print() {
  cout << this->mybool << ' ' << this->myint << endl;
}

void Ref::change(int i) {
  this->myint = i;
}
